pyads/symbol.py

```diff
@@ -0,0 +1,265 @@
+"""
+Define the Symbol class
+
+Separate file because it depends on many other files, so we try to simplify
+the circular dependencies.
+"""
+
+import re
+from ctypes import sizeof
+from typing import TYPE_CHECKING, Any, Optional, List, Tuple, Callable
+from .pyads_ex import adsGetSymbolInfo
+from .structs import NotificationAttrib
+from . import constants  # To access all constants, use package notation
+# ads.Connection relies on structs.AdsSymbol (but in type hints only), so use
+# this 'if' to only include it when type hinting (False during execution)
+if TYPE_CHECKING:
+    from .ads import Connection  # pragma: no cover
+
+
+class AdsSymbol:
+    """Object that points to an ADS variable
+
+    Contains index group, index offset, name, symbol type, comment of ADS
+    symbol. Also remembers a reference to a Connection to be able to
+    read/write directly.
+
+    The virtual property `value` can be used to read from and write to
+    the symbol.
+
+    :ivar index_group: Index group of symbol
+    :ivar index_offset: Index offset of symbol
+    :ivar name: Name of symbol
+    :ivar symbol_type: String representation of symbol type (PLC-style,
+                     e.g. "LREAL")
+    :ivar plc_type: ctypes type of variable (from constants.PLCTYPE_*)
+    :ivar comment: Comment of symbol
+    """
+
+    def __init__(self,
+                 plc: "Connection",
+                 name: Optional[str] = None,
+                 index_group: Optional[int] = None,
+                 index_offset: Optional[int] = None,
+                 symbol_type: Optional[str] = None,
+                 comment=None):
```

Comment:
Return type for __init__ should be None following https://www.python.org/dev/peps/pep-0484/